python/hsfs/core/vector_server.py

```diff
@@ -383,114 +405,124 @@ def _vector_result(self, entry, prepared_statement_objects):
-            for prepared_statement_index in prepared_statement_objects:
-                pk_entry = {}
-                next_statement = False
-                for sk in self._serving_key_by_serving_index[prepared_statement_index]:
-                    if sk.required_serving_key not in entry.keys():
-                        # Check if there is any entry matched with feature name.
-                        if sk.feature_name in entry.keys():
-                            pk_entry[sk.feature_name] = entry[sk.feature_name]
-                        else:
-                            # User did not provide the necessary serving keys, we expect they have
-                            # provided the necessary features as passed_features.
-                            # We are going to check later if this is true
-                            next_statement = True
-                            break
+        bind_entries = []
+        for prepared_statement_index in prepared_statement_objects:
+            pk_entry = {}
+            next_statement = False
+            for sk in self._serving_key_by_serving_index[prepared_statement_index]:
+                if sk.required_serving_key not in entry.keys():
+                    # Check if there is any entry matched with feature name.
+                    if sk.feature_name in entry.keys():
+                        pk_entry[sk.feature_name] = entry[sk.feature_name]
                     else:
-                        pk_entry[sk.feature_name] = entry[sk.required_serving_key]
-                if next_statement:
-                    continue
+                        # User did not provide the necessary serving keys, we expect they have
+                        # provided the necessary features as passed_features.
+                        # We are going to check later if this is true
+                        next_statement = True
+                        break
+                else:
+                    pk_entry[sk.feature_name] = entry[sk.required_serving_key]
+            bind_entries.append(pk_entry)
+            if next_statement:
+                continue
 
-                # Fetch the data from the online feature store
-                prepared_statement = prepared_statement_objects[
-                    prepared_statement_index
-                ]
-                result_proxy = mysql_conn.execute(
-                    prepared_statement, pk_entry
-                ).fetchall()
-                for row in result_proxy:
-                    result_dict = self.deserialize_complex_features(
-                        self._complex_features, row._asdict()
-                    )
-                    serving_vector.update(result_dict)
+        # run all the prepared statements in parallel using aiomysql engine
+        loop = asyncio.get_event_loop()
+        results = loop.run_until_complete(
+            self._execute_prep_statements(
+                list(prepared_statement_objects.values()), bind_entries
```

Comment:
does it guarantee that the order of `prepared_statement_objects.values()` and `bind_entries` are the same?

Reply:
@dhananjay-mk would be good to look at the comments here